node/core/av-store/src/lib.rs

```diff
@@ -145,17 +145,19 @@ async fn run<Context>(subsystem: AvailabilityStoreSubsystem, mut ctx: Context)
 where
 	Context: SubsystemContext<Message=AvailabilityStoreMessage>,
 {
-	let ctx = &mut ctx;
 	loop {
 		select! {
 			incoming = ctx.recv().fuse() => {
 				match incoming {
-					Ok(FromOverseer::Signal(Conclude)) => break,
+					Ok(FromOverseer::Signal(OverseerSignal::Conclude)) => break,
```

Comment:
wait, what?
so the `Conclude` was previously a name binding and rustc didn't warn about the second branch being unreachable?

Reply:
Yup, apparently!

Reply:
oh yeah, also found this in my own branch lol, sorry

Reply:
Did we file a ticket for that? Or did this come as part of a toolchain update?

Reply:
I didn't file a ticket for that. If I had to guess, I'd say that the reason rustc didn't notify us was that this is all inside a macro, and that warning pass happens before macro expansion.

Reply:
> I didn't file a ticket for that. If I had to guess, I'd say that the reason rustc didn't notify us was that this is all inside a macro, and that warning pass happens before macro expansion.

I don't think that's correct. In other places macros are explicitly silenced for unreachable variants.

Reply:
I was wrong. @ordian thanks!

Reply:
You were right @drahnr, it does seem like a bug!

Reply:
Filed https://github.com/rust-lang/rust/issues/78234